client/src/main/kotlin/io/spine/examples/shareaware/client/MarketPage.kt

```diff
@@ -81,6 +109,181 @@ public class MarketPageModel(client: DesktopClient) {
     public fun shares(): StateFlow<AvailableMarketShares?> {
         return sharesSubscriptions.state()
     }
+
+    /**
+     * Sets the page to "purchase" state.
+     *
+     * Page state when the user wants to purchase shares.
+     *
+     * @param share the share to purchase
+     */
+    public fun toPurchaseState(share: Share) {
```

Comment:
Let's speak in terms of domain. E.g. `initiatePurchase` and `complete...`, `cancel...`, etc.